presto-spi/src/main/java/com/facebook/presto/spi/ConnectorPageSink.java

```diff
@@ -14,10 +14,11 @@
 package com.facebook.presto.spi;
 
 import com.facebook.presto.spi.block.Block;
+import io.airlift.slice.Slice;
 
 public interface ConnectorPageSink
 {
     void appendPage(Page page, Block sampleWeightBlock);
 
-    String commit();
+    Iterable<Slice> commit();
```

Comment:
`Set` is definitely wrong because it would be legitimate (albeit strange) for a connector to return duplicate fragments. `List` seems to be over specified because ordering doesn't matter -- they will be merged and passed to the table committer in arbitrary order. The table committer used to take `Collection`, but `Iterable` actually seems preferable because there is no requirement to know the number of items (size).

Is there a flaw in my thinking here?